src/main/java/com/google/jenkins/plugins/k8sengine/KubernetesEngineBuilder.java

```diff
@@ -284,6 +293,48 @@ private boolean verify(
         kubectl, manifestObjects, consoleLogger, verifyTimeoutInMinutes);
   }
 
+  /**
+   * @param credentialsId The service account credential's id.
+   * @return Access token from OAuth to allow kubectl to interact with the cluster.
+   * @throws IOException If an error occurred fetching the access token.
+   */
+  public static String getAccessToken(String credentialsId) throws IOException {
+    Preconditions.checkArgument(!Strings.isNullOrEmpty(credentialsId));
+    HttpTransport HTTP_TRANSPORT = new NetHttpTransport();
+    JsonFactory JSON_FACTORY = new JacksonFactory();
+
+    // get the credential from jenkins
+    List<DomainRequirement> domainRequirements = new ArrayList<>();
+
+    GoogleRobotPrivateKeyCredentials cred =
+        CredentialsMatchers.firstOrNull(
+            CredentialsProvider.lookupCredentials(
+                GoogleRobotPrivateKeyCredentials.class,
+                Jenkins.get(),
+                ACL.SYSTEM,
+                domainRequirements),
+            CredentialsMatchers.withId(credentialsId));
+
+    if (cred != null) {
+      ServiceAccountConfig sac = cred.getServiceAccountConfig();
+
+      GoogleCredential credential =
+          new GoogleCredential.Builder()
+              .setTransport(HTTP_TRANSPORT)
+              .setJsonFactory(JSON_FACTORY)
+              .setServiceAccountId(sac.getAccountId())
+              .setServiceAccountScopes(
+                  ImmutableList.<String>of("https://www.googleapis.com/auth/cloud-platform"))
+              .setServiceAccountPrivateKey(sac.getPrivateKey())
+              .build();
+
+      credential.refreshToken();
+      return credential.getAccessToken();
+    }
+
+    return "";
```

Comment:
Can you add tests for this case?

Reply:
Suggest removing the if statement and eliminating this code path entirely.

Reply:
Can you clarify what you mean by eliminating the code path.

Reply:
If you remove the if block, then this could path will not be necessary. This is related to the feedback I gave re: throwing the error.

Reply:
Done.